nexus/src/external_api/http_entrypoints.rs

```diff
@@ -1010,6 +1030,242 @@ async fn organization_projects_put_project_policy(
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Fetch a single IP Pool.
+#[endpoint {
+    method = GET,
+    path = "/ip-pools/{pool_name}",
+    tags = ["ip-pools"],
+}]
+async fn ip_pools_get_ip_pool(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<IpPoolPathParam>,
+) -> Result<HttpResponseOk<views::IpPool>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let pool_name = &path.pool_name;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let pool = nexus.ip_pool_fetch(&opctx, pool_name).await?;
+        Ok(HttpResponseOk(IpPool::from(pool)))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Delete an IP Pool.
+#[endpoint {
+    method = DELETE,
+    path = "/ip-pools/{pool_name}",
+    tags = ["ip-pools"],
+}]
+async fn ip_pools_delete_ip_pool(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<IpPoolPathParam>,
+) -> Result<HttpResponseDeleted, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let pool_name = &path.pool_name;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        nexus.ip_pool_delete(&opctx, pool_name).await?;
+        Ok(HttpResponseDeleted())
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Update an IP Pool.
+#[endpoint {
+    method = PUT,
+    path = "/ip-pools/{pool_name}",
+    tags = ["ip-pools"],
+}]
+async fn ip_pools_put_ip_pool(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<IpPoolPathParam>,
+    updates: TypedBody<params::IpPoolUpdate>,
+) -> Result<HttpResponseOk<views::IpPool>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let pool_name = &path.pool_name;
+    let updates = updates.into_inner();
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let pool = nexus.ip_pool_update(&opctx, pool_name, &updates).await?;
+        Ok(HttpResponseOk(pool.into()))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+pub type IpPoolRangePaginationParams =
+    PaginationParams<EmptyScanParams, std::net::IpAddr>;
+
+/// List the ranges of IP addresses within an existing IP Pool.
+///
+/// Note that ranges are listed sorted by their first address.
+#[endpoint {
+    method = GET,
+    path = "/ip-pools/{pool_name}/ranges",
+    tags = ["ip-pools"],
+}]
+async fn ip_pool_ranges_get(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<IpPoolPathParam>,
+    // TODO(ben): Figure out how to paginate this
```

Comment:
See my comment in `datastore.rs`. I'm pretty sure that converting from `IpAddr` to `ipnetwork::IpNetwork` makes this doable.